contracts/modules/TransferManager/VolumeRestrictionTM.sol

```diff
@@ -0,0 +1,934 @@
+    // Emit when the new daily (global) restriction is added
+    event AddDailyGlobalRestriction(
+        uint256 _allowedTokens,
+        uint256 _allowedPercentageOfTokens,
+        uint256 _startTime,
+        uint256 _rollingPeriodInDays,
+        uint256 _endTime,
+        uint256 _typeOfRestriction
+    );
+    // Emit when global restriction get modified
+    event ModifyGlobalRestriction(
+        uint256 _allowedTokens,
+        uint256 _allowedPercentageOfTokens,
+        uint256 _startTime,
+        uint256 _rollingPeriodInDays,
+        uint256 _endTime,
+        uint256 _typeOfRestriction
+    );
+    // Emit when daily global restriction get modified
+    event ModifyDailyGlobalRestriction(
+        uint256 _allowedTokens,
+        uint256 _allowedPercentageOfTokens,
+        uint256 _startTime,
+        uint256 _rollingPeriodInDays,
+        uint256 _endTime,
+        uint256 _typeOfRestriction
+    );
+    // Emit when the individual restriction gets removed
+    event IndividualRestrictionRemoved(address _user);
+    // Emit when the global restriction gets removed
+    event GlobalRestrictionRemoved();
+    // Emit when the daily global restriction gets removed
+    event DailyGlobalRestrictionRemoved();
+
+    /**
+     * @notice Constructor
+     * @param _securityToken Address of the security token
+     * @param _polyAddress Address of the polytoken
+     */
+    constructor (address _securityToken, address _polyAddress)
+    public
+    Module(_securityToken, _polyAddress)
+    {
+
+    }
+
+    /**
+     * @notice Used to verify the transfer/transferFrom transaction and prevent tranaction
+     * whose volume of tokens will voilate the maximum volume transfer restriction
+     * @param _from Address of the sender
+     * @param _amount The amount of tokens to transfer
+     * @param _isTransfer Whether or not this is an actual transfer or just a test to see if the tokens would be transferrable
+     */
+    function verifyTransfer(address _from, address /*_to */, uint256 _amount, bytes /*_data*/, bool _isTransfer) public returns (Result) {
+        // If `_from` is present in the exemptionList or it is `0x0` address then it will not follow the vol restriction
+        if (!paused && _from != address(0) && !exemptList[_from]) {
+            // Function must only be called by the associated security token if _isTransfer == true
+            require(msg.sender == securityToken || !_isTransfer);
+            // Checking the individual restriction if the `_from` comes in the individual category 
+            if (individualRestriction[_from].endTime >= now && individualRestriction[_from].startTime <= now) {
+                return _individualRestrictionCheck(_from, _amount, _isTransfer);
+                // If the `_from` doesn't fall under the individual category. It will processed with in the global category automatically
+            } else if (globalRestriction.endTime >= now && globalRestriction.startTime <= now) {
```

Comment:
Yes, Individual Restrictions override global restrictions.

Reply:
Currently, they do, My question was more about the design. 
I feel like they should stack rather than override. We have an exempt list if we want to exempt an individual from global restrictions.
Also, I think the exempt list should only exempt global and daily global restriction. It does not make sense to add an individual restriction and then exempt them from it. Might as well just remove that restriction.

Reply:
An individual restriction is not overriding technically, If `_from` address falls into the individual restriction then it only processed the individual restriction and give the result (no touch to global restriction).  an exempt list is used for exempting the address from all restriction whether it will be global or individual. 
IF I add any address into the Individual restriction then it will automatically be exempted from the global restriction.
If I want to X address will not be affected then I will directly add that address into exemption list. Another case initially X has the individual restriction but it gets expired after some time then that X user automatically falls into the global restriction zone.

Reply:
>An individual restriction is not overriding technically, If _from address falls into the individual restriction then it only processed the individual restriction and give the result (no touch to global restriction).

That's exactly what I meant by overriding. If an individual restriction exist then global restriction is not checked.

>an exempt list is used for exempting the address from all restriction whether it will be global or individual.

As I said earlier about the exempt list, "It does not make sense to add an individual restriction and then exempt them from it. Might as well just remove that restriction."

I understand how the system is currently working, I am proposing that it works differently. I am saying that it will be better if 
1. Both individual and global restrictions are applied by default.
2. Exempt list exempts a user from global restriticon but not individual restriction. If you need to exempt them from individual restriction then just remove that individual restriciton.

Reply:
> That's exactly what I meant by overriding. If an individual restriction exist then global restriction is not checked.
yes, that is happening in the code it doesn't check for the global if individual exists and valid within the time period then it will return without checking the global.

> Exempt list exempts a user from global restriticon but not individual restriction. If you need to exempt them from individual restriction then just remove that individual restriction.

why we are trying to make it complex? if someone is exempted means exempted no restriction on it. If it is not exempted then whether it has the global restriction or individual restriction simple. or do you think there is an advantage from moving to your proposed way?

Reply:
My suggestion does not make it complex, it's just about moving an if statement programmatically.
Logically, my version makes more sense to me. It's about the precedence of things. Individual Restriction should have more priority than an exempt list IMO.

Also, with the current logic, there is no way to apply both global and individual restriction on a person. I think it's an essential feature which can be implemented easily. My way allows you to apply global restriction only/individual restriction only or both restriction simultaneously on any investor.

Reply:
Apologies - I'm not following this argument. Could you give a concrete example using our favourite people Alice & Bob etc.?